corsikaio/subblocks/__init__.py

```diff
@@ -63,3 +62,21 @@ def parse_particle_data(data_block_bytes):
 
 def parse_longitudinal(longitudinal_data_bytes):
     return parse_data_block(longitudinal_data_bytes, longitudinal_data_dtype)
+
+
+def get_units_from_fields(subblock_fields):
+    """Retrieve units a dictionary from the fields of a file subblock.
+
+    Parameters
+    ----------
+    subblock_fields: list(Field)
+        One of the defined lists of fields.
+
+    Return
+    ------
+    units: dict
+        Dictionary with field names as keys and
+        string representations of units as values.
+    """
+    units = {field.name: field.unit for field in subblock_fields}
```

Comment:
Not sure, but do we want to include the fields without units here? Maybe add `if field.unit is not None`?

Reply:
I assumed `units` needs to be as long as a possible `table.colnames` when one does,

`table = Table(data, units={field.name: field.unit for field in fields})`

If this is not true for me it's the same.

`unit` is defined anyway for each Field, fixing a default value makes sense to me

Reply:
No, astropy goes through the dict and attaches units to the columns present in the dict. I think it's good for readability to leave out the `None` units.

Reply:
Will columns without units still be treated as dimensionless quantities?

Reply:
should we keep then the `None` default value in the specification of a new Field?

Reply:
Oh ok I checked it works anyway

Reply:
No, these will be `Columns` in a `QTable`. But there is no difference between putting `name: None` in the dict or not putting anything.

Reply:
```
In [9]: t = QTable({"a": [1, 2, 3]}, units={"a": None})

In [10]: t
Out[10]: 
<QTable length=3>
  a  
int64
-----
    1
    2
    3

In [11]: t["a"]
Out[11]: 
<Column name='a' dtype='int64' length=3>
1
2
3

In [12]: t = QTable({"a": [1, 2, 3]}, units={})

In [13]: t
Out[13]: 
<QTable length=3>
  a  
int64
-----
    1
    2
    3

In [14]: t["a"]
Out[14]: 
<Column name='a' dtype='int64' length=3>
1
2
3

```

Reply:
mmmm I have 2 counter questions:

1. it feels weird to do `get_units_from_fields(some_subblock)` and not see all the original fields (especially if I don't know what am I looking at

2. wouldn't mixed column types mess up with some general operation on a table? if one needs to apply some row operation for example, he/she would have to specify `x.value` on Quantity columns, but not on normal Columns

Reply:
> there is no difference between putting name: None in the dict or not putting anything.

I don't see the same thing:

```python
import numpy as np
from astropy.table import QTable

units = {"b": "cm", "d": "ns"}
units = {"a": None, "b": "cm", "c": None, "d": "ns"}

a = np.array([1, 4, 5], dtype=np.int32)
b = [2.0, 5.0, 8.5]
c = ['x', 'y', 'z']
d = [10, 20, 30]

t = QTable([a, b, c, d], names=('a', 'b', 'c', 'd'), units=units)
t
```

This gives me a Quantity when specifying a dimensionless units, and a Column if not

Reply:
Regarding 1: that was my question... I am inclined that a function called `get_units...` only returns those fields that actually have units.

Regarding 2: that's an issue for astropy. I doubt they will change the `QTable` interface. You can easily check using `if table[col].unit is not None`:

Reply:
Runnig your example I get:

```
In [2]: t["a"]
Out[2]: 
<Column name='a' dtype='int32' length=3>
1
4
5

```

using astropy 5.1